sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,171 @@
+import com.klaviyo.coresdk.model.Event
+import com.klaviyo.coresdk.model.KlaviyoEventType
+import com.klaviyo.coresdk.model.KlaviyoProfileAttributeKey
+import com.klaviyo.coresdk.model.Profile
+import com.klaviyo.coresdk.model.SharedPreferencesDataStore
+import com.klaviyo.coresdk.model.UserInfo
+import com.klaviyo.coresdk.networking.ApiClient
+import com.klaviyo.coresdk.networking.KlaviyoApiClient
+import com.klaviyo.coresdk.networking.KlaviyoNetworkMonitor
+import com.klaviyo.coresdk.networking.NetworkMonitor
 
 /**
- * Our main API for the Klaviyo SDK.
- * Where we pass in analytics requests to be sent to the Klaviyo backend
+ * Public API for the core Klaviyo SDK.
+ * Receives configuration, profile data, and analytics requests
+ * to be processed and sent to the Klaviyo backend
  */
 object Klaviyo {
+    /**
+     * Services registry
+     */
+    object Registry {
+        lateinit var config: Config
+            internal set
+
+        val configBuilder: Config.Builder
+            get() = KlaviyoConfig.Builder()
+
+        var lifecycleMonitor: LifecycleMonitor = KlaviyoLifecycleMonitor
+            private set
+
+        var networkMonitor: NetworkMonitor = KlaviyoNetworkMonitor
+            private set
+
+        var apiClient: ApiClient = KlaviyoApiClient
+            private set
+
+        var dataStore: DataStore = SharedPreferencesDataStore
+            private set
+    }
 
     /**
-     * Assigns a user email address to the user info
-     * This email address is stored in memory to define user details during a session and will be
-     * used to autocomplete analytics requests that are missing the identifier in their property maps
+     * Configure Klaviyo SDK with your account's public API Key and application context.
+     * This must be called to before using any other SDK functionality
      *
-     * Whenever the active user in an app changes (for example via a fresh login) this should be
-     * called to set the new user information in memory.
+     * @param apiKey - Your Klaviyo account's public API Key
+     * @param applicationContext
+     * @return
+     */
+    fun initialize(apiKey: String, applicationContext: Context) {
+        Registry.config = Registry.configBuilder
+            .apiKey(apiKey)
+            .applicationContext(applicationContext)
+            .build()
+    }
+
+    /**
+     * Assigns an email address to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
      *
-     * @param email Email address for an active user
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param email Email address for active user
+     * @return
      */
-    fun setUserEmail(email: String) {
-        UserInfo.email = email
+    fun setEmail(email: String): Klaviyo = apply {
+        this.setProfile(Profile().setEmail(email))
```

Comment:
Possibly dumb question but this looks to me that it initializes a new profile and assigns an email to that. Which would wipe out other profile settings to initialize a new instance (aka: during a login). Does this match up properly with the expectations set by the first line of the docstring for this function though? Its not really the _current_ since if you tried to add an email to an existing profile you'd be resetting it to a new one right?

Am I following that logic correctly?

Reply:
Oh actually sorry I just read how `setProfile` works, but merging info. Makes sense

Reply:
Yeah `Profile` is just a data structure, `UserInfo` is the source of truth for current profile internal state. I chose to make all these setters boil down to using the same method `setProfile`